profiles/coreos/base/profile.bashrc.slsa-provenance

```diff
@@ -0,0 +1,194 @@
+# Build provenance hooks
+# ======================
+# The functions below hook into every ebuild's execution and generate provenance files
+#   to meet the SLSA provenance requirements (https://slsa.dev/spec/v0.1/requirements#available).
+# All input files (source tarball / git commit hash) plus added patches / files,
+#   and all resulting installation binaries and files are captured.
+# The information is emitted in SLSA provenance 0.2 format (see https://slsa.dev/provenance/v0.2)
+
+
+# We only record provenance when a package is actually being built.
+# See profiles/coreos/base/profile.bashrc for cros_...
+cros_post_src_configure_enable_slsa_provenance_report() {
+    if [ "${GENERATE_SLSA_PROVENANCE:-}" != "true" ] ; then
+        einfo "Provenance generation not requested by build; skipping."
+        return 0
+    fi
+    export generate_slsa_provenance_report="yes"
+}
+# --
+
+# Generate SLSA provenance 0.2 Subject information.
+# The information will cover all installation files shipped with a package.
+__slsa_provenance_subject() {
+    local parallel="$(nproc)"
+    local comma=""
+
+    einfo "    Provenance: recording subject (output)"
+
+    echo '   "subject": ['
+
+    (
+        cd "$D"
+        find . -type f -print | sed 's:^./::' | xargs -P "$parallel" -L 1 sha512sum | sort -k2
+    ) | while read checksum filepath; do
+        echo -en "${comma}     {\"name\":\"/$filepath\", \"digest\":{\"sha512\":\"$checksum\"}}"
```

Comment:
This prepends `/` to the "$filepath". I can assume that this results in a proper absolute path. I can't verify it, since build 6068 is already gone from storage.

Reply:
correct, `$filepath` is initially relative to `${D}` so with the prepended `/` the entry becomes absolute. Here's a random part of the glibc report:
```
      {
        "name": "/usr/share/i18n/locales/pa_IN",
        "digest": {
          "sha512": "c7868c23a8f9a48118b150b5327565d4b4f0eca1afcf6061f5068f65a5594b8b17434ec8a16427dd693429a4d43cbb906fc27df6262872dc1199cd83904958c7"
        }
      },
      {
        "name": "/usr/share/i18n/locales/pa_PK",
        "digest": {
          "sha512": "b26b6a0716cdef6c96cda1be9c51b2ee0959e54b8afac0076022a97dae88921e8e94fa25907d01f21dfd57e2daa213f92f5eccf36cb6298e1b8afc340277b856"
        }
      },
```